pkg/operator/ceph/cluster/osd/pod.go

```diff
@@ -90,37 +91,49 @@ func (c *Cluster) makeDeployment(nodeName string, devices []rookalpha.Device, se
 		logger.Infof("no data dir provided")
 	}
 
-	// Mount the path to the directory-based osd unless it is already available under the dataDirHostPath
-	if osd.IsDirectory && !strings.HasPrefix(osd.DataPath, k8sutil.DataDir) {
-		volumeName := k8sutil.PathToVolumeName(osd.DataPath)
-		dataDirSource := v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: osd.DataPath}}
-		volumes = append(volumes, v1.Volume{Name: volumeName, VolumeSource: dataDirSource})
-		volumeMounts = append(volumeMounts, v1.VolumeMount{Name: volumeName, MountPath: osd.DataPath})
-	}
-
-	// Mount the required device
-	if !osd.IsDirectory {
+	// Mount the path to the directory-based osd
+	var dataDir string
+	if osd.IsDirectory {
+		// osd.DataPath includes the osd subdirectory, so we want to mount the parent directory
+		parentDir := filepath.Dir(osd.DataPath)
+		dataDir = parentDir
+		// skip the mount if this is the default directory being mounted, which will look like "/var/lib/rook", even if the dataDirHostPath is a different path
```

Comment:
maybe clarify a little further on this comment that it'll always be `/var/lib/rook` from **inside** the container because `dataDirHostPath` is always mounted to that point inside the container.